src/DbExportsRegistry.cpp

```diff
@@ -0,0 +1,50 @@
+#include "DbExportsRegistry.h"
+
+#include <QDate>
+#include <QSettings>
+
+DbExportsRegistry::DbExportsRegistry(const QString &settingsPath, QObject *parent) :
+    QObject(parent), settingsPath(settingsPath)
+{
+}
+
+QDate DbExportsRegistry::getLastExportDate(const QString &id)
+{
+    QDate lastExport;
+    QSettings s(settingsPath, QSettings::IniFormat);
+    s.sync();
+    s.beginGroup("DbExportRegitry");
+    if (s.contains(id))
+        lastExport = s.value(id).toDate();
+    s.endGroup();
+
+    return lastExport;
+}
+
+bool DbExportsRegistry::isOlderThanAMonth(const QDate &lastExport)
+{
+    QDate limit = QDate::currentDate().addMonths(-1);
+    return lastExport <= limit;
+}
+
+void DbExportsRegistry::setCurrentCardId(const QString &id)
+{
+    DbExportsRegistry::id = id;
+    if (!DbExportsRegistry::id.isEmpty())
+    {
+        QDate date = getLastExportDate(id);
+
+        if  (isOlderThanAMonth(date))
+            emit dbExportRecommended();
```

Comment:
Here only the date is checked. But the db change numbers need also to be checked as user don't need to do a backup if the database has not changed at all during the last month.

Reply:
@limpkin here is the month check

@raoul sorry my bad, we need indeed to do it